api/src/main/java/org/apache/cloudstack/api/command/admin/offering/CreateServiceOfferingCmd.java

```diff
@@ -433,6 +437,10 @@ public Long getStoragePolicy() {
         return storagePolicy;
     }
 
+    public Boolean getDynamicScalingEnabled() {
+        return isDynamicScalingEnabled == null ? Boolean.TRUE : isDynamicScalingEnabled;
```

Comment:
The default value is false, for the global setting "_enable.dynamic.scale.vm_" and parameter "_isdynamicallyscalable_" in registerTemplate/ISO call , defined earlier. Any reason, to make default as "True" here ?

Reply:
1. parameter on template registration is purely a property of a template which defines whether template has hypervisor supported tools installed or not. It is not an option. So default is assumed to be "false". If tools are installed on template then admin will mark it explicitly to "true".
2. default value of global setting is also set to "false" assuming admin will enable the feature if they want overall on CloudStack management zones.
3. When coming to VM or Service offering, these are options to enable or disable at VM level. Given a template is dynamically scalable and global setting is set to true, User can still have option to decide whether he/she wants the VM to dynamically scalable or not. Defaulted to "true" because by if no option is given when deploying the VM then it adheres the template property and global setting by admin.  If we default it to "false" and User did not set any option then even though the template is dynamically scalable and global setting is "true", VM cannot scale.

In summary VM can be scalable only when all options are true. Any parameter setting to "false" will make the VM not dynamically scalable. Each parameter acts as kill switch.

Reply:
Ok @harikrishna-patnala thanks for the detailed explanation. Assuming the same is updated in the CS documentation (wherever required).